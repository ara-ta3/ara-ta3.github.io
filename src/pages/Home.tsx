import React from "react";

import Header from "../components/Header";
import Tools from "../components/Tools";

const Home: React.FC = () => {
  return (
    <>
      <Header />
      <main>
        <Tools />
      </main>
    </>
  );
};

export default Home;
