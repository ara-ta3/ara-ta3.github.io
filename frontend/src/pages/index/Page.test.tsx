import React from "react";
import { render } from "@testing-library/react";
import { expect, test, describe } from "vitest";
import Home from "@/pages/index/+Page.tsx";

describe("Home Snapshot Test", () => {
  test("render Home", () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
