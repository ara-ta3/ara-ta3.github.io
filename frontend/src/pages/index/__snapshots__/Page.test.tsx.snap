// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Snapshot Test > render Home 1`] = `
<DocumentFragment>
  <div
    class="relative flex size-full min-h-screen flex-col bg-white group/design-root overflow-x-hidden"
    style="font-family: "Work Sans", "Noto Sans", sans-serif;"
  >
    <div
      class="layout-container flex h-full grow flex-col"
    >
      <header
        class="flex items-center justify-between whitespace-nowrap border-b border-solid border-b-[#f1f2f4] px-10 py-3"
      >
        <div
          class="flex items-center gap-4 text-[#121416]"
        >
          <div
            class="size-4"
          >
            <svg
              fill="none"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 4H17.3334V17.3334H30.6666V30.6666H44V44H4V4Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <h2
            class="text-[#121416] text-lg font-bold leading-tight tracking-[-0.015em]"
          >
            ara-ta3のポートフォリオ
          </h2>
        </div>
        <div
          class="flex flex-1 justify-end gap-8"
        >
          <div
            class="flex items-center gap-9"
          >
            <a
              class="text-[#121416] text-sm font-medium leading-normal"
              href="#projects"
            >
              プロジェクト
            </a>
            <a
              class="text-[#121416] text-sm font-medium leading-normal"
              href="#articles"
            >
              記事
            </a>
            <a
              class="text-[#121416] text-sm font-medium leading-normal"
              href="#contact"
            >
              SNSアカウント
            </a>
          </div>
        </div>
      </header>
      <div
        class="px-40 flex flex-1 justify-center py-5"
      >
        <div
          class="layout-content-container flex flex-col max-w-[960px] flex-1"
        >
          <div
            class="flex p-4 @container"
          >
            <div
              class="flex w-full flex-col gap-4 items-center"
            >
              <div
                class="flex gap-4 flex-col items-center"
              >
                <div
                  class="bg-center bg-no-repeat aspect-square bg-cover rounded-full min-h-32 w-32"
                  style="background-image: url(https://github.com/ara-ta3.png);"
                />
                <div
                  class="flex flex-col items-center justify-center justify-center"
                >
                  <p
                    class="text-[#121416] text-[22px] font-bold leading-tight tracking-[-0.015em] text-center"
                  >
                    ara-ta3
                  </p>
                  <p
                    class="text-[#6a7581] text-base font-normal leading-normal text-center"
                  >
                    ソフトウェアエンジニア
                  </p>
                  <p
                    class="text-[#6a7581] text-base font-normal leading-normal text-center"
                  >
                    スケーラブルでユーザーフレンドリーなアプリケーションの構築に情熱を持つソフトウェアエンジニアです。React、Scala、ZIOなど様々な技術を使って開発しています。
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            id="projects"
          >
            <h2
              class="text-[#121416] text-[22px] font-bold leading-tight tracking-[-0.015em] px-4 pb-3 pt-5"
            >
              プロジェクト
            </h2>
            <div
              class="grid grid-cols-1 md:grid-cols-2 gap-3 p-4"
            >
              <div
                class="flex flex-col gap-3 pb-3"
              >
                <div
                  class="w-full bg-center bg-no-repeat aspect-video bg-cover rounded-xl"
                  style="background-image: url(https://images.unsplash.com/photo-1518086834179-0d11d2b376c3?w=400&h=300&fit=crop);"
                />
                <div>
                  <p
                    class="text-[#121416] text-base font-medium leading-normal"
                  >
                    Nekometry
                  </p>
                  <p
                    class="text-[#6a7581] text-sm font-normal leading-normal"
                  >
                    猫の体重と活動レベルからRER/DERとごはんの必要な量を計算します。
                  </p>
                </div>
              </div>
              <div
                class="flex flex-col gap-3 pb-3"
              >
                <div
                  class="w-full bg-center bg-no-repeat aspect-video bg-cover rounded-xl"
                  style="background-image: url(https://images.unsplash.com/photo-1461749280684-dccba630e2f6?w=400&h=300&fit=crop);"
                />
                <div>
                  <p
                    class="text-[#121416] text-base font-medium leading-normal"
                  >
                    個人ウェブサイト
                  </p>
                  <p
                    class="text-[#6a7581] text-sm font-normal leading-normal"
                  >
                    React、Vike、TypeScriptを使用した静的サイト生成。
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            id="articles"
          >
            <h2
              class="text-[#121416] text-[22px] font-bold leading-tight tracking-[-0.015em] px-4 pb-3 pt-5"
            >
              記事
            </h2>
            <div
              class="flex items-center gap-4 bg-white px-4 min-h-[72px] py-2 justify-between"
            >
              <div
                class="flex flex-col justify-center"
              >
                <p
                  class="text-[#121416] text-base font-medium leading-normal line-clamp-1"
                >
                  TypeScript + Vike + SSG Getting Started
                </p>
                <p
                  class="text-[#6a7581] text-sm font-normal leading-normal line-clamp-2"
                >
                  Vikeを使った静的サイト生成について
                </p>
              </div>
              <div
                class="shrink-0"
              >
                <a
                  href="https://zenn.dev/ara_ta3/articles/typescript-vike-ssg-getting-started"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <div
                    class="text-[#121416] flex size-7 items-center justify-center"
                    data-icon="ArrowRight"
                    data-size="24px"
                    data-weight="regular"
                  >
                    <svg
                      fill="currentColor"
                      height="24px"
                      viewBox="0 0 256 256"
                      width="24px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M221.66,133.66l-72,72a8,8,0,0,1-11.32-11.32L196.69,136H40a8,8,0,0,1,0-16H196.69L138.34,61.66a8,8,0,0,1,11.32-11.32l72,72A8,8,0,0,1,221.66,133.66Z"
                      />
                    </svg>
                  </div>
                </a>
              </div>
            </div>
            <div
              class="flex items-center gap-4 bg-white px-4 min-h-[72px] py-2 justify-between"
            >
              <div
                class="flex flex-col justify-center"
              >
                <p
                  class="text-[#121416] text-base font-medium leading-normal line-clamp-1"
                >
                  猫のカロリー計算ツールの開発
                </p>
                <p
                  class="text-[#6a7581] text-sm font-normal leading-normal line-clamp-2"
                >
                  ペットの健康管理アプリケーション開発記
                </p>
              </div>
              <div
                class="shrink-0"
              >
                <a
                  href="https://arata.hatenadiary.com/entry/2025/01/09/174000"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <div
                    class="text-[#121416] flex size-7 items-center justify-center"
                    data-icon="ArrowRight"
                    data-size="24px"
                    data-weight="regular"
                  >
                    <svg
                      fill="currentColor"
                      height="24px"
                      viewBox="0 0 256 256"
                      width="24px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M221.66,133.66l-72,72a8,8,0,0,1-11.32-11.32L196.69,136H40a8,8,0,0,1,0-16H196.69L138.34,61.66a8,8,0,0,1,11.32-11.32l72,72A8,8,0,0,1,221.66,133.66Z"
                      />
                    </svg>
                  </div>
                </a>
              </div>
            </div>
            <div
              class="flex items-center gap-4 bg-white px-4 min-h-[72px] py-2 justify-between"
            >
              <div
                class="flex flex-col justify-center"
              >
                <p
                  class="text-[#121416] text-base font-medium leading-normal line-clamp-1"
                >
                  動的ルーティングでのSSG実装
                </p>
                <p
                  class="text-[#6a7581] text-sm font-normal leading-normal line-clamp-2"
                >
                  メタ情報を含む静的サイト生成のノウハウ
                </p>
              </div>
              <div
                class="shrink-0"
              >
                <a
                  href="https://zenn.dev/ara_ta3/articles/typescript-vike-ssg-with-dynamic-routing"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <div
                    class="text-[#121416] flex size-7 items-center justify-center"
                    data-icon="ArrowRight"
                    data-size="24px"
                    data-weight="regular"
                  >
                    <svg
                      fill="currentColor"
                      height="24px"
                      viewBox="0 0 256 256"
                      width="24px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M221.66,133.66l-72,72a8,8,0,0,1-11.32-11.32L196.69,136H40a8,8,0,0,1,0-16H196.69L138.34,61.66a8,8,0,0,1,11.32-11.32l72,72A8,8,0,0,1,221.66,133.66Z"
                      />
                    </svg>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            id="contact"
          >
            <h2
              class="text-[#121416] text-[22px] font-bold leading-tight tracking-[-0.015em] px-4 pb-3 pt-5"
            >
              SNSアカウント
            </h2>
            <div
              class="@container"
            >
              <div
                class="gap-2 px-4 flex flex-wrap justify-start"
              >
                <a
                  href="https://x.com/ara_ta3"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <div
                    class="flex flex-col items-center gap-2 bg-white py-2.5 text-center w-20"
                  >
                    <div
                      class="rounded-full bg-[#f1f2f4] p-2.5"
                    >
                      <div
                        class="text-[#121416]"
                        data-icon="SocialIcon"
                        data-size="20px"
                        data-weight="regular"
                      >
                        <svg
                          fill="currentColor"
                          height="20px"
                          viewBox="0 0 256 256"
                          width="20px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M247.39,68.94A8,8,0,0,0,240,64H209.57A48.66,48.66,0,0,0,168.1,40a46.91,46.91,0,0,0-33.75,13.7A47.9,47.9,0,0,0,120,88v6.09C79.74,83.47,46.81,50.72,46.46,50.37a8,8,0,0,0-13.65,4.92c-4.31,47.79,9.57,79.77,22,98.18a110.93,110.93,0,0,0,21.88,24.2c-15.23,17.53-39.21,26.74-39.47,26.84a8,8,0,0,0-3.85,11.93c.75,1.12,3.75,5.05,11.08,8.72C53.51,229.7,65.48,232,80,232c70.67,0,129.72-54.42,135.75-124.44l29.91-29.9A8,8,0,0,0,247.39,68.94Zm-45,29.41a8,8,0,0,0-2.32,5.14C196,166.58,143.28,216,80,216c-10.56,0-18-1.4-23.22-3.08,11.51-6.25,27.56-17,37.88-32.48A8,8,0,0,0,92,169.08c-.47-.27-43.91-26.34-44-96,16,13,45.25,33.17,78.67,38.79A8,8,0,0,0,136,104V88a32,32,0,0,1,9.6-22.92A30.94,30.94,0,0,1,167.9,56c12.66.16,24.49,7.88,29.44,19.21A8,8,0,0,0,204.67,80h16Z"
                          />
                        </svg>
                      </div>
                    </div>
                    <p
                      class="text-[#121416] text-sm font-medium leading-normal"
                    >
                      X (Twitter)
                    </p>
                  </div>
                </a>
                <a
                  href="https://github.com/ara-ta3"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <div
                    class="flex flex-col items-center gap-2 bg-white py-2.5 text-center w-20"
                  >
                    <div
                      class="rounded-full bg-[#f1f2f4] p-2.5"
                    >
                      <div
                        class="text-[#121416]"
                        data-icon="SocialIcon"
                        data-size="20px"
                        data-weight="regular"
                      >
                        <svg
                          fill="currentColor"
                          height="20px"
                          viewBox="0 0 256 256"
                          width="20px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M208.31,75.68A59.78,59.78,0,0,0,202.93,28,8,8,0,0,0,196,24a59.75,59.75,0,0,0-48,24H124A59.75,59.75,0,0,0,76,24a8,8,0,0,0-6.93,4,59.78,59.78,0,0,0-5.38,47.68A58.14,58.14,0,0,0,56,104v8a56.06,56.06,0,0,0,48.44,55.47A39.8,39.8,0,0,0,96,192v8H72a24,24,0,0,1-24-24A40,40,0,0,0,8,136a8,8,0,0,0,0,16,24,24,0,0,1,24,24,40,40,0,0,0,40,40H96v16a8,8,0,0,0,16,0V192a24,24,0,0,1,48,0v40a8,8,0,0,0,16,0V192a39.8,39.8,0,0,0-8.44-24.53A56.06,56.06,0,0,0,216,112v-8A58.14,58.14,0,0,0,208.31,75.68ZM200,112a40,40,0,0,1-40,40H112a40,40,0,0,1-40-40v-8a41.74,41.74,0,0,1,6.9-22.48A8,8,0,0,0,80,73.83a43.81,43.81,0,0,1,.79-33.58,43.88,43.88,0,0,1,32.32,20.06A8,8,0,0,0,119.82,64h32.35a8,8,0,0,0,6.74-3.69,43.87,43.87,0,0,1,32.32-20.06A43.81,43.81,0,0,1,192,73.83a8.09,8.09,0,0,0,1,7.65A41.72,41.72,0,0,1,200,104Z"
                          />
                        </svg>
                      </div>
                    </div>
                    <p
                      class="text-[#121416] text-sm font-medium leading-normal"
                    >
                      GitHub
                    </p>
                  </div>
                </a>
                <a
                  href="https://zenn.dev/ara_ta3"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <div
                    class="flex flex-col items-center gap-2 bg-white py-2.5 text-center w-20"
                  >
                    <div
                      class="rounded-full bg-[#f1f2f4] p-2.5"
                    >
                      <div
                        class="text-[#121416]"
                        data-icon="SocialIcon"
                        data-size="20px"
                        data-weight="regular"
                      >
                        <svg
                          fill="currentColor"
                          height="20px"
                          viewBox="0 0 256 256"
                          width="20px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M224,48H32a8,8,0,0,0-8,8V192a16,16,0,0,0,16,16H216a16,16,0,0,0,16-16V56A8,8,0,0,0,224,48ZM203.43,64,128,133.15,52.57,64ZM216,192H40V74.19l82.59,75.71a8,8,0,0,0,10.82,0L216,74.19V192Z"
                          />
                        </svg>
                      </div>
                    </div>
                    <p
                      class="text-[#121416] text-sm font-medium leading-normal"
                    >
                      Zenn
                    </p>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="flex justify-center"
      >
        <div
          class="flex max-w-[960px] flex-1 flex-col"
        >
          <div
            class="flex flex-col gap-6 px-5 py-10 text-center @container"
          >
            <p
              class="text-[#6a7581] text-base font-normal leading-normal"
            >
              © 2024 ara-ta3. All rights reserved.
            </p>
          </div>
        </div>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
